libsolidity/interface/StandardCompiler.cpp

```diff
@@ -131,6 +132,11 @@ Json::Value formatErrorWithException(
 		formatSourceLocation(boost::get_error_info<errinfo_sourceLocation>(_exception)),
 		formatSecondarySourceLocation(boost::get_error_info<errinfo_secondarySourceLocation>(_exception))
 	);
+
+	if (_errorId.error != 0)
+		error["id"] = static_cast<unsigned>(_errorId.error);
```

Comment:
errorid?

Reply:
"id" is an attribute of an error, just like "type" of "message". (I mean, they are not called "errorType" or "errorMessage".)

Reply:
I was mainly suggesting 'errorid' because 'id' is an extremely generic term that often has a special meaning in such structures which might suggest that it uniquely identifies the object (the reported error, not the kind of error) inside the json.